pkg/monitoring/metrics/virt-controller/vmistats_collector.go

```diff
@@ -73,6 +74,14 @@ var (
 		},
 		[]string{"node", "namespace", "name"},
 	)
+
+	vmiAddresses = operatormetrics.NewGaugeVec(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_vmi_status_addresses",
+			Help: "The addresses of a VirtualMachineInstance.",
+		},
+		[]string{"node", "namespace", "name", "ip", "mac", "host_interface_name", "vmi_interface_name", "info_source"},
```

Comment:
@machadovilaca Hi, Are you not missing "type" for InternalIP , ExternalIP, Hostname ?
Please explain what the "host_interface_name", "vmi_interface_name" and "info_source" mean and check that the labels names are the same as in other related metrics, if applicable.

Also, I see that "kube_node_status_addresses" reports the status. Do we also keep the vmi addresses status and can report it here?

Reply:
> Are you not missing "type" for InternalIP , ExternalIP, Hostname ?

VMIs unlike nodes don't have the internal/external IP differentiation.
The hostname is not reported in VirtualMachineInstanceNetworkInterface.

> Please explain what the "host_interface_name", "vmi_interface_name" and "info_source" mean

**host_interface_name** is the name of the network interface in the host node that's being linked to the VM
**vmi_interface_name** is the name of the previous interface insided the VM
**info_source** specifies the origin of the data collected if from the domain, guest-agent or multus-status

> check that the labels names are the same as in other related metrics, if applicable

no other occurrences yet

> Also, I see that "kube_node_status_addresses" reports the status. Do we also keep the vmi addresses status and can report it here?

They only report node, type, and address as labels
The status refers to the data source node.status and here vmi.status

Reply:
can we add also a label to see the pod ip's the vmi running on? something like pod_internal_ip pode_external_ip.
also can we add also maybe region?

Reply:
we should correlate that info with other metrics if needed, doesn't make sense here imo